docs/resources/bms_instance.md

```diff
@@ -102,18 +102,27 @@ The following arguments are supported:
 * `nics` - (Required, List, ForceNew) Specifies an array of one or more networks to attach to the instance. The network
   object structure is documented below. Changing this creates a new instance.
 
-* `admin_pass` - (Optional, String, ForceNew) Specifies the administrative password to assign to the instance. Changing
-  this creates a new instance.
-
-* `key_pair` - (Optional, String, ForceNew) Specifies the name of a key pair to put on the instance. The key pair must
-  already be created and associated with the tenant's account. Changing this creates a new instance.
+* `admin_pass` - (Optional, String, ForceNew) Specifies the administrative password to assign to the instance when
+  create BMS with Password mode. Changing this creates a new instance. The use of this field has the following
+  restrictions:
+  + The valid length of this field is 8 to 26 characters.
+  + The value can contain only uppercase letters, digits, and special characters (!@%^-_=+[{}]:,./?) and must contain
+    at least three of the preceding character types.
+  + The value cannot contain the username or the username spelled backwards.
+
+* `key_pair` - (Optional, String, ForceNew) Specifies the name of a key pair to put on the instance when create BMS with
+  Key pair mode. The key pair must already be created and associated with the tenant's account. The field is required
+  when using a Windows image to create a BMS. Changing this creates a new instance.
 
 * `user_data` - (Optional, String, ForceNew) Specifies the user data to be injected during the instance creation. Text
   and text files can be injected. `user_data` can come from a variety of sources: inline, read in from the
-  *file* function. Changing this creates a new instance.
+  *file* function. Changing this creates a new instance. The use of this field has the following restrictions:
+  + The content to be injected can be encoded with base64 or plain text.
+  + The maximum size of the content to be injected (before encoding) is 32 KB.
 
--> **NOTE:** If the `user_data` field is specified for a Linux BMS that is created using an image with Cloud-Init
-installed, the `admin_pass` field becomes invalid.
+-> **NOTE:** The value of `admin_pass` or `user_data` will be injected into the BMS as a password when create BMS with
+  Password mode. If both `key_pair` and `user_data` are specified, `user_data` only injects user data when create BMS
+  with Key pair mode.
```

Comment:
-> **NOTE:** (1) If the `user_data` field is specified for a Linux BMS that is created using an image with Cloud-Init
installed, the `admin_pass` field becomes invalid. (2) The field `admin_pass` is valid only when `key_pair` and `user_data`
are not specified. (3) The field `key_pair` is valid only when `admin_pass` is not specified. (4) The three parameters
`admin_pass`, `key_pair` and `user_data` cannot all be empty. (5) If both `key_pair` and `user_data` are specified,
`user_data` only injects user data.